src/resmom/linux/alps.c

```diff
@@ -6397,129 +6668,923 @@ alps_inventory(void)
-	},
-	{
-		BASIL_ELM_RELEASED,
-		released_start,
-		default_element_end,
-		disallow_char_data
-	},
-	{
-		BASIL_ELM_INVENTORY,
-		inventory_start,
-		inventory_end,
-		disallow_char_data
-	},
-	{
-		BASIL_ELM_ENGINE,
-		engine_start,
-		default_element_end,
-		disallow_char_data
-	},
-	{
-		BASIL_ELM_NODEARRAY,
-		node_array_start,
-		default_element_end,
-		disallow_char_data
-	},
-	{
-		BASIL_ELM_NODE,
-		node_start,
-		default_element_end,
-		disallow_char_data
-	},
-	{
-		BASIL_ELM_SOCKETARRAY,
-		socket_array_start,
-		default_element_end,
-		disallow_char_data
-	},
-	{
-		BASIL_ELM_SOCKET,
-		socket_start,
-		default_element_end,
-		disallow_char_data
-	},
-	{
-		BASIL_ELM_SEGMENTARRAY,
-		segment_array_start,
-		default_element_end,
-		disallow_char_data
-	},
-	{
-		BASIL_ELM_SEGMENT,
-		segment_start,
-		default_element_end,
-		disallow_char_data
-	},
-	{
-		BASIL_ELM_CUARRAY,
-		computeunit_array_start,
-		default_element_end,
-		disallow_char_data
-	},
-	{
-		BASIL_ELM_COMPUTEUNIT,
-		computeunit_start,
-		default_element_end,
-		disallow_char_data
-	},
-	{
-		BASIL_ELM_PROCESSORARRAY,
-		processor_array_start,
-		default_element_end,
-		disallow_char_data
-	},
-	{
-		BASIL_ELM_PROCESSOR,
-		processor_start,
-		default_element_end,
-		disallow_char_data
-	},
-	{
+	/*
+	 * Determine if ALPS supports the BASIL 1.7 protocol. We are only
+	 * partially supporting the BASIL 1.7 protocol (for the System Query).
+	 */
+	alps_engine_query_KNL();
+
+	if (basil_1_7_supported)
+		log_event(PBSEVENT_DEBUG4, PBS_EVENTCLASS_NODE, LOG_DEBUG, __func__,
+			"This Cray system supports the BASIL 1.7 protocol.");
+	else {
+		log_event(PBSEVENT_DEBUG4, PBS_EVENTCLASS_NODE, LOG_ERR, __func__,
+			"This Cray system does not support the BASIL 1.7 protocol.");
+		return;
+	}
+
+	/*
+	 * Allocate a buffer (requestBuffer_knl) for a new ALPS request (System Query).
+	 * A nonzero return value indicates failure; return at this point without proceeding
+	 * with System Query processing.
+	 */
+	if (init_KNL_alps_req_buf() != 0)
+		return;
+
+	/* Create a System (BASIL 1.7) Query request to fetch KNL information. */
+	snprintf(requestBuffer_knl, UTIL_BUFFER_LEN, "<?xml version=\"1.0\"?>\n"
+		"<" BASIL_ELM_REQUEST " "
+		BASIL_ATR_PROTOCOL "=\"%s\" "
+		BASIL_ATR_METHOD "=\"" BASIL_VAL_QUERY "\" "
+		BASIL_ATR_TYPE "=\"" BASIL_VAL_SYSTEM "\"/>", basilversion_system);
+
+	/*
+	 * The 'basil_ver' argument is checked in response_start() (a callback
+	 * function invoked during alps_request() processing). Flow of control can
+	 * arrive at response_start() from either alps_system_KNL() or alps_inventory().
+	 * This argument helps make the distinction.
+	 */
+	if ((brp_knl = alps_request(requestBuffer_knl, basilversion_system)) == NULL) {
+
+		/* Failure to get KNL Node information from ALPS. */
+		log_event(PBSEVENT_SYSTEM, PBS_EVENTCLASS_NODE, LOG_NOTICE, __func__,
+			  "ALPS System Query request failed.");
+		return;
+	}
+
+	if (basil_system_knl != NULL)
```

Comment:
This is where you set the variable but there is no use for it. Please remove this block if it is not needed.

Reply:
Done